aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -185,6 +208,40 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+
+    public entry fun rotate_authentication_key_ed25519(account: &signer, signature: vector<u8>, new_public_key: vector<u8>) acquires Account, OriginatingAddress {
```

Comment:
Add /// comments at the top of this function explaining what it does.

Reply:
Explain what is 'signature': what is the message signed, and under what secret key is it being signed?